rg_instructor_analytics_log_collector/tests/processors/test_processor.py

```diff
@@ -0,0 +1,95 @@
+"""
+Test the `Processor` module.
+"""
+import logging
+from unittest import TestCase
+
+from ddt import ddt, data, unpack
+from mock import patch
+
+from rg_instructor_analytics_log_collector.processors.processor import Processor
+from rg_instructor_analytics_log_collector.processors.base_pipeline import BasePipeline
+from rg_instructor_analytics_log_collector.processors.student_step_pipeline import StudentStepPipeline
+
+
+class TestRecords(object):
```

Comment:
I do not understand why this class is needed, could you help me understand?

Reply:
Apart from being an iterator, the class encapsulates additional business logic which facilitates objects mocking (as discussed).